iceoryx_examples/waitset/ice_waitset_gateway.cpp

```diff
@@ -74,8 +83,10 @@ int main()
         auto& subscriber = subscriberVector.back();
 
         waitset
-            .attachEvent(
-                subscriber, iox::popo::SubscriberEvent::DATA_RECEIVED, 0, createEventCallback(subscriberCallback))
+            .attachEvent(subscriber,
+                         iox::popo::SubscriberEvent::DATA_RECEIVED,
+                         0,
+                         createEventCallback(subscriberCallback, sumOfAllSamples))
```

Comment:
sumOfAllSamples is used across threads so it should be atomic. I know it works in general, but is not guaranteed to in theory. Note that concurrent updates are not the problem since the listener just has one thread (but this would not need to be the case from the interface only). However, the update may never reach the main thread like this without being atomic.